src/beacons/use-maybe-show-share-travel-habits-screen.tsx

```diff
@@ -25,25 +18,23 @@ export const useMaybeShowShareTravelHabitsScreen = (
   } = useRemoteConfig();
   const sessionCountRef = useRef(0);
   const isInitializedRef = useRef(false);
+  const {isBeaconsSupported, kettleInfo} = useBeacons();
 
   const appStatus = useAppStateStatus();
 
   const {onboarded} = useAppState();
-  const [isBeaconsEnabled, isBeaconsEnabledDebugOverrideReady] =
-    useIsBeaconsEnabled();
   const [hasSeenShareTravelHabitsScreen] = useHasSeenShareTravelHabitsScreen();
   const enabled =
     onboarded &&
-    isBeaconsEnabledDebugOverrideReady &&
-    isBeaconsEnabled &&
+    isBeaconsSupported &&
     hasSeenShareTravelHabitsScreen !== null &&
     !hasSeenShareTravelHabitsScreen;
 
   const maybeShowShareTravelHabitsScreen = useCallback(async () => {
-    if (await shouldShowShareTravelHabitsScreen()) {
+    if (!kettleInfo?.isBeaconsOnboarded) {
```

Comment:
This will now show even if `hasLocationWhenInUsePermission` is false.
If I remember correctly, you can't get locationAlways permission without first getting locationWhenInUse permission on iOS?

Reply:
Btw should it also check that `hasSeenShareTravelHabitsScreen` is `false` here?